push

```diff
@@ -54,31 +64,63 @@ def _controlled_tempdir():
     finally:
         shutil.rmtree(td)
 
-def _build_fw(zip_path, apps_path):
-    with ZipFile(zip_path, 'w') as zf:
-        for fname in os.listdir(apps_path):
-            zf.write(os.path.join(apps_path, fname), fname)
-
-def _transfer_firmware(host, repo_path, scp, ssh, sensors):
+def _build_fw(zip_path, apps_path, targets):
+    if targets:
+        regex_list = [re.compile(f"{target}" + r"(.*).hex") for target in targets]
+        with ZipFile(zip_path, 'w') as zf:
+            for fname in os.listdir(apps_path):
```

Comment:
Same as the above comment, the `scripts/subsystem_versions.py` script validates the filenames, it could be useful to check that.